scalameta/contrib/shared/src/main/scala/scala/meta/contrib/AssociatedComments.scala

```diff
@@ -59,16 +59,22 @@ object AssociatedComments {
         if (isLeading) leading += c
         else trailing += c
       case Token.LF() => isLeading = true
+      case Token.EOF() =>
+        val l = leading.result()
+        val t = trailing.result()
+        if (l.nonEmpty || t.nonEmpty) {
+          trailingBuilder += lastToken -> (l ::: t)
+        }
```

Comment:
this could also be implemented outside of the `foreach`. Please let me know if you prefer that instead